src/ts/universe.ts

```diff
@@ -47,16 +55,29 @@ export class UniverseObject {
   }
 }
 
+interface UniverseObjectWithMassOptions extends UniverseObjectOptions {
+  mass?: number;
+}
+
 /** An object with mass, but does not require that it takes up space. E.g. black holes */
 export class UniverseObjectWithMass extends UniverseObject {
   mass: number;
 
-  constructor(x: number, y: number) {
-    super(x, y);
-    this.mass = 0;
+  constructor(options: UniverseObjectWithMassOptions) {
+    super(options);
+    this.mass = options.mass ?? 0;
```

Comment:
I'd just use `||` here, `??` transpiles down to `options.mass !== null && options.mass !== undefined ? options.mass : 0`

Reply:
The nice thing is, we aren't transpiling so we get the literal `??` in the output, so it doesn't take up more space.